src/script/common/c_content.cpp

```diff
@@ -1564,6 +1585,127 @@ ToolCapabilities read_tool_capabilities(
 	return toolcap;
 }
 
+/******************************************************************************/
+PointabilityType read_pointability(lua_State *L, int index)
+{
+	if (lua_isboolean(L, index)) {
+		if (lua_toboolean(L, index))
+			return POINTABLE;
+		else
+			return POINTABLE_NOT;
+	} else if (lua_isstring(L, index)) {
+		size_t len;
+		const char* s = lua_tolstring(L, index, &len);
+		if (!strcmp(s, "blocking"))
+			return POINTABLE_BLOCKING;
+	}
+	return POINTABLE;
+}
+
+/******************************************************************************/
+PointingAbilities read_pointabilities(lua_State *L, int index)
+{
+	PointingAbilities pointabilities;
+	
+	lua_getfield(L, index, "nodes");
+	if(lua_istable(L, -1)){
+		int ti = lua_gettop(L);
+		lua_pushnil(L);
+		while(lua_next(L, ti) != 0) {
+			// key at index -2 and value at index -1
+			std::string name = luaL_checkstring(L, -2);
+			
+			// handle groups
+			if(name.find("group:") != std::string::npos) {
+				pointabilities.node_groups[name.substr(6)] = read_pointability(L, -1);
+			} else {
+				pointabilities.nodes[name] = read_pointability(L, -1);
+			}
+			
+			// removes value, keeps key for next iteration
+			lua_pop(L, 1);
+		}
+	}
+	lua_pop(L, 1);
+	
+	lua_getfield(L, index, "objects");
+	if(lua_istable(L, -1)){
+		int ti = lua_gettop(L);
+		lua_pushnil(L);
+		while(lua_next(L, ti) != 0) {
+			// key at index -2 and value at index -1
+			std::string name = luaL_checkstring(L, -2);
+			
+			// handle groups
+			if(name.find("group:") != std::string::npos) {
+				pointabilities.object_groups[name.substr(6)] = read_pointability(L, -1);
+			} else {
+				pointabilities.objects[name] = read_pointability(L, -1);
+			}
+			
+			// removes value, keeps key for next iteration
+			lua_pop(L, 1);
+		}
+	}
+	lua_pop(L, 1);
+	
+	return pointabilities;
+}
+
+/******************************************************************************/
+void push_pointability(lua_State *L, const PointabilityType& pointable)
+{
+	switch(pointable)
+	{
+	  case POINTABLE:
+		lua_pushboolean(L, true);
+		break;
+	  case POINTABLE_NOT:
+		lua_pushboolean(L, false);
+		break;
+	  case POINTABLE_BLOCKING:
+		lua_pushlstring(L, "blocking", 8);
```

Comment:
```suggestion
		lua_pushliteral(L, "blocking");
```

Reply:
fixed

Reply:
It's `lua_pushstring` now.
`lua_pushliteral` is for string literals, see <https://www.lua.org/manual/5.1/manual.html#lua_pushliteral>.

Reply:
Sorry, I missed this, it is lua_pushliteral now.

(Everywhere else inside the file lua_pushstring has been used for string literals, so maybe in the old code it should also be changed, but I guess it isn't very important.)